src/internal/storage/server.go

```diff
@@ -135,6 +143,35 @@ func (s *Server) CreateFileset(server storage.Fileset_CreateFilesetServer) error
 	})
 }
 
+func (s *Server) copyFile(ctx context.Context, uw *fileset.UnorderedWriter, msg *storage.CopyFile) error {
+	id, err := fileset.ParseID(msg.FilesetId)
+	if err != nil {
+		return err
+	}
+	fs, err := s.Filesets.Open(ctx, []fileset.ID{*id})
+	if err != nil {
+		return err
+	}
+	srcPath := pfsfile.CleanPath(msg.Src)
+	dstPath := srcPath
+	if msg.Dst != "" {
+		dstPath = pfsfile.CleanPath(msg.Dst)
+	}
+	fs = fileset.NewIndexFilter(fs, func(idx *index.Index) bool {
+		return idx.Path == srcPath || strings.HasPrefix(idx.Path, fileset.Clean(srcPath, true))
```

Comment:
I thought we had a function for this somewhere.

Reply:
I don't think so.